controllers/utils/utils.go

```diff
@@ -61,3 +69,50 @@ func MinRequeueDuration(old, new *time.Duration) *time.Duration {
 	}
 	return old
 }
+
+// GetAllRemediationTemplates returns a slice of all ObjectReferences used as RemedediationTemplate in the
+// given NodeHealthCheck
+func GetAllRemediationTemplates(nhc *v1alpha1.NodeHealthCheck) []*v1.ObjectReference {
+	refs := make([]*v1.ObjectReference, 0)
+	if nhc.Spec.RemediationTemplate != nil {
```

Comment:
NIT: Similarly to my above comment, there is no nill protection for the object `nhc`, prior to accessing its fields (e.g., `nhc.Spec.RemediationTemplate`), here and in the next function `GetRemediationDuration`. Probably a little bit overkill check to do, but I wanted to raise that at least.

Reply:
If you want to develop in a very defensive way, yes, then you should check every argument if it's valid.
In this case it's more than unlikely that nhc is nil, because the NHC CR is the starting point of basically everything we do. In practice I'm even sure that today it can never be nil.